File: Programmers/Lv_1/기사단원의무기.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cmath>
#include <map>
using namespace std;

int divisor_cnt(int n) {
    int cnt = 0;
    for (int i = 1; i <= sqrt(n); i++) {
        if (n % i == 0) {
            cnt++;
            if (n / i != i) cnt++;
        }
    }
    return cnt;
}

int solution(int number, int limit, int power) {
    int answer = 0;
    
    map<int, int> m;
    
    for (int i = 1; i <= number; i++) {
        m.insert({i, divisor_cnt(i)});
    }

    for (auto iter: m) {
        if (iter.second > limit) {
            answer += power;
            continue;
        }
        answer += iter.second;
    }

    return answer;
}

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int number, limit, power;
    cin >> number >> limit >> power;

    cout << solution(number, limit, power);

    return 0;
}

File: Programmers/Lv_1/기사단원의무기.py
def solution(number, limit, power):
    answer = 0
    divisorCount = []
    for i in range(1, number+1):
        if getDivisorCount(i) > limit:
            divisorCount.append(power)
        else:
            divisorCount.append(getDivisorCount(i))
    
    answer = sum(divisorCount)
    return answer

def getDivisorCount(n):
    divisorsList = []
    for i in range(1, int(n**(1/2)) + 1):
        if (n % i == 0):
            divisorsList.append(i)
            if ((i**2) != n):
                divisorsList.append(n // i)
    
    return len(divisorsList)


def cf(n):
    a = []
    for i in range(1, int(n**0.5) + 1):
        if n % i == 0:
            a.append(n // i)
            a.append(i)
            
    return len(set(a))

def solution2(number, limit, power):
    return sum([cf(i) if cf(i) <= limit else power for i in range(1, number+1)])